### PR TITLE
Added vehicle support for safe zone

### PR DESCRIPTION
**Pull request by JohanLanges**
Firing from vehicles (infantry) is still possible...

Firing the turrets are disabled.

Doesn't apply to helicopters and jets who're flying.

Most likely needs refactoring. But I've got a headache at the moment, so here it is.

Will implement the message system soon.

### CODE REVIEW

**Comment by legman** [comment]
This isn't quite what i had in mind.

a) I dont like the second function
b) it adds event listeners to vehicles and doesnt clean up fully

My idea is you add and remove an event listener every you switch vehicle in the zone, then you only need to track one and you arnt recieveing events for vehicles you are no longer driving.

The vehicle protection should be built into the activateSafeZone so it piggy backs on the existing trigger.

We will have a proper discussion later in teamspeak on how you should set up the while loop!

**Comment by JohanLanges** [comment]
Firing from vehicles is now disabled
Firing a turret is disabled
Added the event handler (Fired) when spawning a vehicle
Made safeZones a global variable. This variable needs to exist! But can be just an empty array.
Multiple safe zones is supported

To do: Exclude SOS members from safe zone. Although I don't know if that is the best thing to do...

**Comment by legman** [comment]
What you have done is much better - good job!

There are a few things you need to change though, first when declaring a public variable we will use a certain style - 

``` javascript
SOS_BASE_SAFE_ZONES;
```

Also we need to move required functionality from fn_init so inside fn_addSafeZone you can do this

``` javascript
if (isNil "SOS_BASE_SAFE_ZONES") then {
    SOS_BASE_SAFE_ZONES = [];
};
SOS_BASE_SAFE_ZONES pushBack "triggerName";

blah blah...
```

this means we just add then in fn_init and it will define the global array needed if it does not exist already.

File: SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf [hunk 1]
```diff
@@ -31,3 +31,15 @@ _vehicle addEventHandler ["GetIn", {
         };
     };
 }];
+
+_vehicle addEventHandler ["Fired", {
+	{
+		_inArea = [getPos player, _x] call CBA_fnc_inArea;
```
Comment (legman) [inline SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf]: getPosATL

z axis is relative to the terrain

getPos is global position

File: SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf [hunk 2]
```diff
@@ -31,3 +31,15 @@ _vehicle addEventHandler ["GetIn", {
...
+		_flying = getPos player select 2 > 5;
```
Comment (legman) [inline SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf]: **getPosATL**

z axis is relative to the terrain

**getPos** is global position

File: SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf [hunk 3]
```diff
@@ -31,3 +31,15 @@ _vehicle addEventHandler ["GetIn", {
         };
     };
 }];
+
```
Comment (legman) [inline SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf:34]: comment please _// checks if vehicle fires inside a safe zone_

File: SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf [hunk 4]
```diff
@@ -31,3 +31,15 @@ _vehicle addEventHandler ["GetIn", {
...
+_vehicle addEventHandler ["Fired", {
+	{
+		_inArea = [getPos player, _x] call CBA_fnc_inArea;
+		_flying = getPos player select 2 > 5;
+		if(_inArea && !_flying) then {
+			deleteVehicle (_this select 6);
```
Comment (legman) [inline SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf]: line 41, 42 could be in a function **fn_showWarning**

File: SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf (hunk 3 above)
Comment (legman) [inline SpecialOperations2.Altis/sos/vehicle/functions/fn_initVehicle.sqf:34]:
``` javascript
if !([] call SOS_fnc_isMember) then {
    // listen for vehicle event  
};
```

**Comment by legman** [comment]
can you sort out your indentations please - other than that and the few changes, PERFECTO!

will merge when finished

**Pull request merged**